src/core/entities.rs

```diff
@@ -1,26 +1,58 @@
 #[cfg_attr(rustfmt, rustfmt_skip)]
-#[derive(Debug, Clone, Deserialize, Serialize, PartialEq)]
+#[derive(Debug, Clone, PartialEq)]
 pub struct Entry {
+    pub id             : String,
+    pub osm_node       : Option<u64>,
+    pub created        : u64,
+    pub version        : u64,
+    pub title          : String,
+    pub description    : String,
+    pub location       : Location,
+    pub contact        : Option<Contact>,
+    pub homepage       : Option<String>,
+    pub categories     : Vec<String>,
+    pub tags           : Vec<String>,
+    pub license        : Option<String>,
+    pub image_url      : Option<String>,
+    pub image_link_url : Option<String>,
+}
+
+#[cfg_attr(rustfmt, rustfmt_skip)]
+#[derive(Debug, Clone, PartialEq)]
+pub struct Location {
+    pub lat     : f64,
+    pub lng     : f64,
+    pub address : Option<Address>
+}
+
+#[cfg_attr(rustfmt, rustfmt_skip)]
+#[derive(Debug, Clone, PartialEq)]
+pub struct Address {
```

Comment:
Both `Address` and `Contact` are suitable for deriving `Default`

Reply:
:+1: